bib/internal/uploader/aws.go

```diff
@@ -32,26 +47,22 @@ func doUpload(a AwsUploader, file *os.File, bucketName, keyName string, pbar *pb
 		if err != nil {
 			return nil, fmt.Errorf("cannot stat upload: %v", err)
 		}
-		pbar.SetTotal(st.Size())
-		pbar.Set(pb.Bytes, true)
-		pbar.SetWriter(osStdout)
-		r = pbar.NewProxyReader(file)
-		pbar.Start()
-		defer pbar.Finish()
+		pbar.SetMessagef("Uploading %s to %s", file.Name(), bucketName)
+		r = &proxyReader{0, file, pbar, 0, st.Size()}
 	}
 
 	return a.UploadFromReader(r, bucketName, keyName)
 }
 
-func UploadAndRegister(a AwsUploader, filename, bucketName, imageName, targetArch string, pbar *pb.ProgressBar) error {
+func UploadAndRegister(a AwsUploader, filename, bucketName, imageName, targetArch string, pbar progress.ProgressBar) error {
 	file, err := os.Open(filename)
 	if err != nil {
 		return fmt.Errorf("cannot upload: %v", err)
 	}
 	defer file.Close()
 
 	keyName := fmt.Sprintf("%s-%s", uuid.New().String(), filepath.Base(filename))
-	fmt.Fprintf(osStdout, "Uploading %s to %s:%s\n", filename, bucketName, keyName)
+	pbar.SetMessagef("Uploading %s to %s:%s\n", filename, bucketName, keyName)
```

Comment:
Can we have a separate message for the registering step? See lines 76-79. Note that the AMI id should be printed to stdout, so people can use bib in scripts. Other messages probably don't belong to stdout.